lightning/src/onion_message/messenger.rs

```diff
@@ -246,6 +246,20 @@ pub trait CustomOnionMessageHandler {
 	fn read_custom_message<R: io::Read>(&self, message_type: u64, buffer: &mut R) -> Result<Option<Self::CustomMessage>, msgs::DecodeError>;
 }
 
+/// An processed incoming onion message, containing either a Forward (another onion)
+/// or a Receive payload with decrypted contents
+pub enum PeeledOnion<CMH: Deref> where
+	CMH::Target: CustomOnionMessageHandler,
+{
+	/// Forwarded onion, with the next node id and a new onion
+	Forward(PublicKey, msgs::OnionMessage),
+	/// Received onion message, with decrypted contents, path_id, and reply path
+	Receive(OnionMessageContents<<<CMH as Deref>::Target as CustomOnionMessageHandler>::CustomMessage>, Option<[u8; 32]>, Option<BlindedPath>)
+}
+
+/// Errors that may occur when [receiving an onion message].
+#[derive(Debug, PartialEq, Eq)]
+pub struct ReceiveError { }
```

Comment:
I don't think we should bother adding a struct for this if it's gonna be empty, we should just `return Err(())` in `peel_onion` instead. If we wanted to be fancy we could error with an enum that has a variant for each possible error, but that seems like overkill here?

Reply:
makes sense